pkg/streamingpromql/optimize/plan/narrow_selectors_test.go

```diff
@@ -202,6 +202,38 @@ func TestNarrowSelectorsOptimizationPass(t *testing.T) {
 							- MatrixSelector: {__name__="some_metric"}[5m0s]
 			`,
 		},
+		// Make sure we don't modify query plans that modify labels
+		"binary expression with label_replace on one side": {
+			expr: `sum by (statefulset) (kube_statefulset_replicas) - sum by (statefulset) (label_replace(not_ready, "statefulset", "$1", "job", ".+/(.+)"))`,
```

Comment:
What if the expression was something like this?

```promql
sum by (env, region) (first_metric) - sum by (env, region) (label_replace(second_metric, "region", "$1", "job", ".*/(.*)"))
```

In that case, it'd be safe to pass a matcher for the `env` label.